cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/detector/notifier/SelfHealingNotifierTest.java

```diff
@@ -43,83 +45,111 @@ public void testOnBrokerFailure() {
     assertEquals(1, result.delay());
-    assertFalse(anomalyNotifier.alertCalled);
+    assertFalse(anomalyNotifier._alertCalled.get(AnomalyType.BROKER_FAILURE));
 
     // Sleep 1 ms
     mockTime.sleep(1);
-    anomalyNotifier.resetAlert();
+    anomalyNotifier.resetAlert(AnomalyType.BROKER_FAILURE);
     result = anomalyNotifier.onBrokerFailure(new BrokerFailures(mockKafkaCruiseControl, failedBrokers, allowCapacityEstimation));
     assertEquals(AnomalyNotificationResult.Action.CHECK, result.action());
     assertEquals(SelfHealingNotifier.DEFAULT_AUTO_FIX_THRESHOLD_MS + failureTime1 - mockTime.milliseconds(),
                  result.delay());
-    assertTrue(anomalyNotifier.alertCalled);
+    assertTrue(anomalyNotifier._alertCalled.get(AnomalyType.BROKER_FAILURE));
 
     // Sleep to 1 ms before alert.
     mockTime.sleep(result.delay() - 1);
-    anomalyNotifier.resetAlert();
+    anomalyNotifier.resetAlert(AnomalyType.BROKER_FAILURE);
     result = anomalyNotifier.onBrokerFailure(new BrokerFailures(mockKafkaCruiseControl, failedBrokers, allowCapacityEstimation));
     assertEquals(AnomalyNotificationResult.Action.CHECK, result.action());
     assertEquals(1, result.delay());
-    assertTrue(anomalyNotifier.alertCalled);
-    assertFalse(anomalyNotifier.autoFixTriggered);
+    assertTrue(anomalyNotifier._alertCalled.get(AnomalyType.BROKER_FAILURE));
+    assertFalse(anomalyNotifier._autoFixTriggered.get(AnomalyType.BROKER_FAILURE));
 
     // Sleep 1 ms
     mockTime.sleep(1);
-    anomalyNotifier.resetAlert();
+    anomalyNotifier.resetAlert(AnomalyType.BROKER_FAILURE);
     result = anomalyNotifier.onBrokerFailure(new BrokerFailures(mockKafkaCruiseControl, failedBrokers, allowCapacityEstimation));
     assertEquals(AnomalyNotificationResult.Action.FIX, result.action());
     assertEquals(-1L, result.delay());
-    assertTrue(anomalyNotifier.alertCalled);
-    assertTrue(anomalyNotifier.autoFixTriggered);
+    assertTrue(anomalyNotifier._alertCalled.get(AnomalyType.BROKER_FAILURE));
+    assertTrue(anomalyNotifier._autoFixTriggered.get(AnomalyType.BROKER_FAILURE));
+    assertFalse(anomalyNotifier._alertCalled.get(AnomalyType.GOAL_VIOLATION));
+    assertFalse(anomalyNotifier._alertCalled.get(AnomalyType.METRIC_ANOMALY));
   }
 
   @Test
   public void testSelfHealingDisabled() {
-    final long failureTime1 = 200L;
-    final long failureTime2 = 400L;
     final long startTime = 500L;
     Time mockTime = new MockTime(startTime);
     KafkaCruiseControl mockKafkaCruiseControl = EasyMock.mock(KafkaCruiseControl.class);
     TestingBrokerFailureAutoFixNotifier anomalyNotifier = new TestingBrokerFailureAutoFixNotifier(mockTime);
-    anomalyNotifier.configure(Collections.singletonMap(SelfHealingNotifier.SELF_HEALING_ENABLED_CONFIG, "false"));
 
+    Map<String, String> selfHealingExplicitlyDisabled = new HashMap<>(3);
+    selfHealingExplicitlyDisabled.put(SelfHealingNotifier.SELF_HEALING_BROKER_FAILURE_ENABLED_CONFIG, "false");
```

Comment:
I think we can set `SelfHealingNotifierSELF_HEALING_ENABLED_CONFIG` to `True` here to verify the overriding of  specific config over general config.